github.go

```diff
@@ -1088,3 +1088,61 @@ func (c *Client) SearchRepos(query string) ([]*github.Repository, error) {
 
 	return allRepos, nil
 }
+
+func (c *Client) SearchCode(query string) ([]*github.Repository, error) {
+
+	client := c.client
+
+	opt := &github.SearchOptions{
+		ListOptions: github.ListOptions{PerPage: 100},
+	}
+	// get all pages of results
+	var allRepos []*github.Repository
+	for {
+		var repos *github.CodeSearchResult
+		var resp *github.Response
+		errs := RetryExponentialBackoff(9999, time.Second, func() error {
+			var err error
+
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+			defer cancel()
+
+			repos, resp, err = client.Search.Code(ctx, query, opt)
+			if err != nil {
+				return fmt.Errorf("error while executing request: %s", err)
+			}
+			ResponseCallback(resp)
+			if handleRateLimitError(err, resp) {
+				return err
+			}
+
+			if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNotFound && resp.StatusCode != http.StatusNoContent {
+				// TODO: catch rate limit error, and wait
+				return fmt.Errorf(
+					"status code is: %v (%s)",
+					resp.StatusCode,
+					resp.Status,
+				)
+			}
+			// nil on 200 and 404
+			return nil
+		})
+		if errs != nil && len(errs) > 0 {
+			return nil, errors.New(FormatErrorArray("", errs))
+		}
+		if resp.StatusCode == http.StatusNotFound {
+			// TODO: catch rate limit error, and wait
+			return nil, ErrNotFound
+		}
+
+		for repIndex := range repos.CodeResults {
+			allRepos = append(allRepos, &repos.CodeResults[repIndex].Repository) // to get the actual Repository, not CodeResult ¿?
```

Comment:
I have doubts on this one, but I think the rest are OK. Seen [this](https://github.com/google/go-github/blob/a64f3a6d084e5a38c26b8cac35101d47981298e3/github/search.go) :)